Sources/Misc/SystemInfo.swift

```diff
@@ -261,6 +261,21 @@ extension SystemInfo {
         #endif
     }
 
+    static var applicationDidBecomeActiveNotification: Notification.Name? {
+        #if os(iOS) || os(tvOS) || VISION_OS || targetEnvironment(macCatalyst)
+            return UIApplication.didBecomeActiveNotification
+        #elseif os(macOS)
+            return NSApplication.didBecomeActiveNotification
+        #elseif os(watchOS)
+            if #available(watchOS 7.0, *) {
+                return WKApplication.didBecomeActiveNotification
+            } else {
+                // There's no equivalent notification available on watchOS <7.
```

Comment:
We'll probably need to document the limitation of macOS (does not become active automatically after purchase) and watchOS (only supported for 7.0 and later, while we support 6.2 and later for the SDK)

Reply:
Agreed! We should document it in the DocC comments for this property in case it's used for something else in the future.

I don't think it'll be an issue for SK2 observer mode on watchOS - wasn't StoreKit2 first available with watchOS8?

Reply:
Did you try if we can use [NSExtensionHostDidBecomeActiveNotification](https://developer.apple.com/documentation/foundation/nsextensionhostdidbecomeactivenotification?language=objc) for this use case?

Reply:
Yeah,I tried that first before `WKApplication.didBecomeActiveNotification`. Unfortunately it was never fired during the IAP flow so I had to move on to something else. 🤷‍♂️

Reply:
FWIW, IAPs were introduced in watchOS 6.2, and in a fairly broken state (the `appStoreReceiptURL` even points to the wrong file, we added our own workaround for that), with very few apps using it in watchOS < 7. So I wouldn't really stress too much about that limitation.

Reply:
I wouldn't even mind bumping the requirement for watchOS to 7.0, really.